src/engine/BMInterfaceTournament.php

```diff
@@ -0,0 +1,987 @@
+        if (is_null($tournament)) {
+            $this->set_message("Tournament $tournId does not exist");
+            return NULL;
+        }
+
+        if ($tournament->tournamentState > BMTournamentState::JOIN_TOURNAMENT) {
+            $this->set_message('The tournament has already started.');
+            return NULL;
+        }
+
+        if ($tournament->creatorId !== $userId) {
+            $this->set_message('You did not create this tournament.');
+            return NULL;
+        }
+
+        return TRUE;
+    }
+
+    /**
+     * Attempt to cancel a tournament in the database
+     *
+     * @param int $userId
+     * @param int $tournId
+     * @return bool
+     */
+    protected function cancel_tournament_in_database($userId, $tournId) {
+        // query is written in this way to avoid possible race conflicts between
+        // multiple people trying to act on a tournament at the same time
+        //
+        // a tournament can only be deleted by the creator if it has not started yet
+        $query = 'UPDATE tournament '.
+                 'SET status_id = (SELECT id FROM tournament_status WHERE name = "CANCELLED"), '.
+                 'tournament_state = :tournament_state '.
+                 'WHERE id = :tourn_id '.
+                 'AND tournament_state <= 20 '.
+                 'AND creator_id = :creator_id';
+        $parameters = array(':tourn_id' => $tournId,
+                            ':tournament_state' => BMTournamentState::CANCELLED,
+                            ':creator_id' => $userId);
+
+        return self::$db->update_and_report_if_changed($query, $parameters);
+    }
+
+    /**
+     * Dismiss tournament link from overview page
+     *
+     * @param int $playerId
+     * @param int $tournId
+     * @return bool
+     */
+    public function dismiss_tourn($playerId, $tournId) {
+        try {
+            $query1 =
+                'SELECT s.name AS "status" ' .
+                'FROM tournament AS t ' .
+                'INNER JOIN tournament_status AS s ON s.id = t.status_id ' .
+                    'LEFT JOIN tourn_player_map AS m ' .
+                    'ON m.tourn_id = t.id AND m.player_id = :player_id ' .
+                'WHERE t.id = :tourn_id';
+            $parameters = array(
+                ':player_id' => $playerId,
+                ':tourn_id' => $tournId,
+            );
+            $columnReturnTypes = array(
+                'status' => 'str',
+            );
+            $rows = self::$db->select_rows($query1, $parameters, $columnReturnTypes);
+
+            if (count($rows) == 0) {
+                $this->set_message("Tournament $tournId does not exist");
+                return NULL;
+            }
+            if (($rows[0]['status'] != 'COMPLETE') &&
+                ($rows[0]['status'] != 'CANCELLED')) {
+                $this->set_message("Tournament $tournId isn't complete");
+                return NULL;
+            }
```

Comment:
You could simplify this code by using `self::$db->select_single_value()` and catching `BMDatabaseException` for the "tournament does not exist" case.

Reply:
Code simplified as suggested.